### ISSUE TITLE
Cy.log args are no longer printed to the console showing newlines after cypress 7.2

### ISSUE THREAD

**Issue by westhomas** [cypress-io/cypress#17542]
### Current behavior

`cy.log` shows args as an array and doesn't show the newlines that may be present. This make it very difficult to read large debug messages (something my QA users have come to depend on when debugging tests).

### Desired behavior

`cy.log` entries when clicked are printed to the developer console showing newlines. This is helpful if you have large debug logs you wish to view. This is how it used to work in 7.2 and prior. 

### Test code to reproduce

```
"cypress": "7.2.0",  <=== shows newlines in console logs
...
"cypress": "7.3.0",  <=== behavior changed
```

```
it('cy.log works', () => {
  cy.log("Hello there from javascript", "this is not an array\nlarge logs with lots of lines\n\nmore logging\n\neven more")
})
```

### Cypress Version

7.3+

### Other

Cypress 7.2 and earlier:
<img width="925" alt="Screen Shot 2021-07-30 at 3 24 39 PM" src="https://user-images.githubusercontent.com/953369/127708357-b5b84606-865f-424f-9f41-dca80cadc6b1.png">

Cypress 7.3 and later:
<img width="925" alt="Screen Shot 2021-07-30 at 3 23 45 PM" src="https://user-images.githubusercontent.com/953369/127708361-8c0f4abe-09f0-4138-adc9-1787bc746f9c.png">

**Comment by westhomas** [comment]
This is the commit responsible. And it make sense:
https://github.com/cypress-io/cypress/commit/439bbfd88b87d3ce61ea7d39ce2befc5cf0f39d4

Would it be appropriate to change line 43 to spread args as well?

```
Cypress.log({
    end: true,
    snapshot: true,
    message: [msg, ...args],
    consoleProps() {
        return {
            message: msg,
-            args,
+            ...args,
        }
    },
})
```

I believe that would yield better results.

<img width="1120" alt="Screen Shot 2021-07-30 at 4 19 02 PM" src="https://user-images.githubusercontent.com/953369/127712557-aa8a8d33-044c-4cfc-bd21-f314c0d0d666.png">

Demo code:

```
const logCustom = (msg, ...args) => {
    Cypress.log({
        end: true,
        snapshot: true,
        message: [msg, ...args],
        consoleProps() {
            return {
                message: msg,
                ...args,
            }
        },
    })
}

it('cy.log works', () => {
    cy.log("Hello there from javascript", "this is not an array\nlarge logs with lots of lines\n\nmore logging\n\neven more")

    logCustom("logCustom:Hello there from javascript", "this is not an array\nlarge logs with lots of lines\n\nmore logging\n\neven more")

    cy.log('s', 1, 2, 3, 4)

    logCustom('logCustom:s', 1, 2, 3, 4)
})
```

I can do a PR if someone wants to sanction this change.

**Comment by jennifer-shehane** [comment]
Interesting. Thanks for the thorough description and for tracking down the commit. cc @sainthkh

**Comment by sainthkh** [comment]
@westhomas I tried your solution, but it didn't solve the problem and makes the output look unintuitive when there are 2+ args like below:

![Screenshot from 2021-08-09 09-56-36](https://user-images.githubusercontent.com/8130013/128653602-8e782a8c-0b8c-4cf6-a21d-b4bd4b7d1a01.png)

It's hard to guess what `0`, `1` means.

So, I changed it this way:

![](https://user-images.githubusercontent.com/8130013/128653481-1469d838-b15a-442b-93a6-6ede370055dd.png)

**Comment by chrisbreiding** [comment]
Released in [`8.4.0`](https://on.cypress.io/changelog#8-4-0).

This comment thread has been locked. If you are still experiencing this issue after upgrading to
Cypress v8.4.0, please open a new issue.
